src/init.cpp

```diff
@@ -1035,6 +1039,37 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
         }
     }
 
+    if (!mapMultiArgs["-nuparams"].empty()) {
+        // Allow overriding network upgrade parameters for testing
+        if (!Params().MineBlocksOnDemand()) {
```

Comment:
Is there not a more perspicuous way to check for regtest?

Reply:
This is how Bitcoin Core does it 🤷‍♂️ We could probably also use one of the internal constants to check against; upstream at some point also moved to using strings for identifying parameter types, which might also be useful, but is out of scope for this PR.

Reply:
File a ticket for this (e.g. adding an ``IsRegtest()`` method to ``CBaseChainParams``, or whatever seems best).

Reply:
Forgot we can also use `Params().NetworkIDString() != "regtest"`, switching to that.